### PR TITLE
Paint Dodgems

### PR DESCRIPTION
**Pull request by marijnvdwerf**
image

### CODE REVIEW

**Comment by marijnvdwerf** [comment]
@duncanspumpkin @IntelOrca Another paint function. I've tried to break this one up more.

File: src/ride/track_paint.c [hunk 1]
```diff
@@ -982,6 +982,257 @@ TRACK_PAINT_FUNCTION get_track_paint_function_topspin(int trackType, int directi
...
+
+		case 13:
+		case 14:
+			dodgems_paint_sub_fence_top_left(rideIndex, height, mapElement);
+			break;
+
+		case 15:
+			dodgems_paint_sub_fence_top_left(rideIndex, height, mapElement);
+			dodgems_paint_sub_fence_bottom_left(rideIndex, height, mapElement);
+			break;
```
Comment (janisozaur) [inline src/ride/track_paint.c]: what about 5,9,10?

Comment (duncanspumpkin) [inline src/ride/track_paint.c]: They probably don't exist

Comment (duncanspumpkin) [inline src/ride/track_paint.c]: Or they don't need a fence

File: src/ride/track_paint.c [hunk 2]
```diff
@@ -982,6 +982,257 @@ TRACK_PAINT_FUNCTION get_track_paint_function_topspin(int trackType, int directi
...
+	RCT2_GLOBAL(0x141E9C8, uint16) = (height & 0xFF00) | 0x20;
+	RCT2_GLOBAL(0x141E9CC, uint16) = (height & 0xFF00) | 0x20;
+	RCT2_GLOBAL(0x141E9D0, uint16) = (height & 0xFF00) | 0x20;
+	RCT2_GLOBAL(0x141E9D4, uint16) = (height & 0xFF00) | 0x20;
+
+	height += 12;
+	if (RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_PAINT_TILE_MAX_HEIGHT, sint16) < height) {
+		RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_PAINT_TILE_MAX_HEIGHT, sint16) = height;
+		RCT2_GLOBAL(0x141E9DA, uint8) = 32;
+	}
```
Comment (IntelOrca) [inline src/ride/track_paint.c]: This if statement is common for a lot of paint functions, so we could turn this into a simple helper.

File: src/ride/track_paint.c (hunk 1 above)
Comment (marijnvdwerf) [inline src/ride/track_paint.c]: image

**Pull request merged**